nsxt/resource_nsxt_upgrade_run.go

```diff
@@ -503,9 +523,76 @@ func updateUpgradeUnitGroups(upgradeClientSet *upgradeClientSet, d *schema.Resou
 	for _, groupI := range d.Get(componentToGroupKey[component]).([]interface{}) {
 		group := groupI.(map[string]interface{})
 		groupID := group["id"].(string)
-		groupGet, err := upgradeClientSet.GroupClient.Get(groupID, nil)
-		if err != nil {
-			return err
+		var groupGet *model.UpgradeUnitGroup
+		var err error
+		isCreate := false
+		if groupID == "" {
+			groupName := group["display_name"].(string)
+			if groupName == "" {
+				return fmt.Errorf("couldn't find upgrade unit group without id or display_name")
+			}
+			// This is a custom group, try to find it by name
+			groupList, err := upgradeClientSet.GroupClient.List(nil, nil, nil, nil, nil, nil, nil, nil)
```

Comment:
is pagination relevant here?

Reply:
Do we expect more than 1000 groups?

Reply:
Doesn't sound feasible, but maybe a comment about it would be helpful